inference-engine/src/gna_plugin/transformations/remove_single_input_concat.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <openvino/cc/ngraph/itt.hpp>
+
+#include "transformations/remove_single_input_concat.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset7.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/pass/manager.hpp>
+
+using NodeInput = ngraph::Input<ngraph::Node>;
+using NodeOutput = ngraph::Output<ngraph::Node>;
+
+namespace GNAPluginNS {
+    NGRAPH_RTTI_DEFINITION(RemoveSingleInputConcat, "RemoveSingleInputConcat", 0);
+
+    RemoveSingleInputConcat::RemoveSingleInputConcat() {
+        MATCHER_SCOPE(RemoveSingleInputConcat);
+
+        auto concat_operation = ngraph::pattern::wrap_type<ngraph::opset7::Concat>();
+
+        ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+            const auto& pattern_map = m.get_pattern_value_map();
+            const auto concat_operation_node = pattern_map.at(concat_operation).get_node_shared_ptr();
+
+            if (concat_operation_node->get_input_size() != 1 || !concat_operation_node->get_output_size())
```

Comment:
It is better to move this check number of inputs into the pattern to avoid extra callback calls. Also, I think that check of outputs number is excessive.